compiler/src/dotty/tools/dotc/transform/BetaReduce.scala

```diff
@@ -76,5 +81,13 @@ object BetaReduce:
       substTo = argSyms
     ).transform(ddef.rhs)
 
-    seq(bindings.result(), expansion)
+    val expansion1 = new TreeMap {
+      override def transform(tree: Tree)(using Context) = tree.tpe.widenTermRefExpr match
+        case ConstantType(const) if isPureExpr(tree) => cpy.Literal(tree)(const)
+        case _ => super.transform(tree)
+    }.transform(expansion)
```

Comment:
This is constant folding -- can we reuse the logic in `ConstFold`?

Reply:
It is not exactly constant folding. It is just the propagation of constants. After this `ConstFold` is used at some point and performs the actual folding.

Reply:
What about `TreeInfo.constToLiteral`?

Reply:
It fails. It also looks like overkill.

Reply:
OK. Maybe as an optimization for later, beta-reduce & inlining may create new const-folding opportunities.

Reply:
Indeed